File: java/Tiger.java
public class Tiger extends Animal implements Comparable<Tiger> {
    private Continent continent;
    public static final Continent DEFAULT_CONTINENT = Continent.UNKNOWN;

    public Tiger() {
        super();
        this.continent = Continent.UNKNOWN;
    }

    public Tiger(String name) {
        this(name, DEFAULT_BIRTH, 0.0, DEFAULT_GENDER, DEFAULT_CONTINENT);
    }

    public Tiger(String name, Date birth, double weight, Continent continent) {
        this(name, birth, weight, DEFAULT_GENDER, continent);
    }

    public Tiger(String name, Date birth, double weight, Gender gender, Continent continent) {
        super(name, birth, weight, gender);
        this.continent = continent;
    }

    public Continent getContinent() {
        return continent;
    }

    public void setContinent(Continent continent) {
        this.continent = continent;
    }

    @Override
    public void move() {
        System.out.println("Tiger " + getName() + " is moving.");
    }

    @Override
    public void utter() {
        System.out.println("Tiger " + getName() + " is roaring.");
    }

    @Override
    public String toString() {
        return "Name: " + getName() + ", Birth: " + getBirth() + ", Weight: " + getWeight() +
                "kg, Gender: " + getGender() + ", Continent: " + continent;
    }

    @Override
    public int compareTo(Tiger other) {
        return this.getName().compareTo(other.getName());
    }
}

File: cpp/Tiger.cpp
#include "Tiger.h"

#include <cmath>
#include <iostream>
#include <sstream>

Tiger::Tiger()
    : Animal("", Date(), 0.0, ::Gender::Other), continent(Continent::Unknown) {}

Tiger::Tiger(std::string name)
    : Animal(name, Date(), 0.0, ::Gender::Other),
      continent(Continent::Unknown) {}

Tiger::Tiger(std::string name, Date birth, double weight, ::Continent continent)
    : Animal(name, birth, weight, ::Gender::Other), continent(continent) {}

Tiger::Tiger(std::string name, Date birth, double weight, ::Gender gender,
             ::Continent continent)
    : Animal(name, birth, weight, gender), continent(continent) {}

std::string Tiger::ToString() const {
  std::stringstream ss;
  ss << "Name: " << name << ", Birth " << birth.ToString()
     << ", Weight: " << weight << "kg"
     << ", Gender: " << gender << ", Continent: " << continent;
  return ss.str();
}
void Tiger::move() {
  std::cout << "Tiger " << name << " is moving." << std::endl;
}

void Tiger::utter() {
  std::cout << "Tiger " << name << " is roaring." << std::endl;
}